synapse/handlers/auth.py

```diff
@@ -1200,7 +1200,26 @@ def generate_oidc_session_token(
         nonce: str,
         client_redirect_url: str,
         duration_in_ms: int = (60 * 60 * 1000),
-    ):
+    ) -> str:
+        """Generates a signed token storing data about an OIDC session.
+
+        When Synapse initiates an authorization flow, it creates a random state
+        and a random nonce. Those parameters are given to the provider and
+        should be verified when the client comes back from the provider.
+        It is also used to store the client_redirect_url, which is used to
+        complete the SSO login flow.
+
+        Args:
+            state (str): The ``state`` parameter passed to the OIDC provider.
+            nonce (str): The ``nonce`` parameter passed to the OIDC provider.
+            client_redirect_url (str): The URL the client gave when it
+                initiated the flow.
+            duration_in_ms (int): An optional duration for the token in milliseconds.
+                Defaults to an hour.
+
+        Returns:
+            str: A signed macaroon token with the session informations.
```

Comment:
Listing the types in both the docstring and the function signature is redundant. My understanding is that the type hints are preferred.

Reply:
Yes, please list them just as type hints (we're slowly transitioning the code as we touch pieces of it).